run_tests.py

```diff
@@ -155,30 +156,36 @@ def prompt_user_to_run_tests(test_names):
         elif answer.upper()[0] == "Y":
             return
 
+def runTestProto(self):
```

Comment:
move this in the class

Reply:
rename to fail_with_message

Reply:
I originally had it as a class mehtod, but it didn't like it as you have to pass the name of the function to super init and it seemed to think it didn't exist at that point. But maybe you have to do something cleverer with the syntax?

Reply:
I mean pass it to setattr so that you can then pass the name to super init

Reply:
Well, it now seems to work as a class method - maybe I had setattr wrong first time or something